src/reducer.d.ts

```diff
@@ -0,0 +1,32 @@
+export type ReducerName =
+  | "first"
+  | "last"
+  | "count"
+  | "distinct"
+  | "sum"
+  | "proportion"
+  | "proportion-facet"
+  | "deviation"
+  | "min"
+  | "min-index"
+  | "max"
+  | "max-index"
+  | "mean"
+  | "median"
+  | "variance"
+  | "mode"
+  | `p${number}${number}` // percentile
```

Comment:
This allows something like "p1.51.5" where 1.5 and 1.5 each satisfy `number`. If we want to just allow numbers between 00 and 99 we can change number to Digit and define Digit as `type Digit = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9;`

Reply:
This avoids listing the 100 pXX combination. Ideally we'd have a way to give two examples (say, p05 and p95), cover p00...p99, but without expanding to p00…p99 in the autocomplete.

This works for me in VSCode:

```diff
-  | `p${number}${number}` // percentile
+  | "p05"
+  | "p95"
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  | (`p${Digit}${Digit}` & String) // percentile
```

Reply:
You shouldn’t use `String` as a type here (hence the ESLint warning). You need to use `Record<never, never>` instead. (And that’s only because of https://github.com/microsoft/TypeScript/issues/29729.)

Reply:
Good call, I've edited my suggestion. And it works!
<img width="323" alt="Capture d’écran 2023-03-13 à 15 58 12" src="https://user-images.githubusercontent.com/7001/224740373-3e5acd64-7846-4646-a3b1-a5716229ccd4.png">

Reply:
> This avoids listing the 100 pXX combination.

We don’t allow `p100` at runtime. I think that’s fine; you can use `max` instead.

https://github.com/observablehq/plot/blob/5830098d729edb56bde0b60f281c2e744919d5fa/src/transforms/group.js#L244

Reply:
Making Duane’s fix causes VS Code to suggest every possible percentile… 😂 I might suppress these using the Record hack.

<img width="465" alt="Screenshot 2023-03-13 at 8 08 26 AM" src="https://user-images.githubusercontent.com/230541/224743198-67337289-896f-48b9-be71-37188632c68b.png">

Reply:
Yep! This is what @Fil was suggesting above, to manually include a few common percentiles and then suppress the rest